src/client/resources/common.ts

```diff
@@ -50,3 +56,41 @@ export abstract class ClusterResource {
 function namedPath(listPath, name) {
   return [listPath, name].join('/');
 }
+
+export abstract class ClusterDiscoveryResource {
+
+  private _type: string;
+  private _cluster: string;
+
+  public abstract discoveryParameters(): IDiscoveryParametrs;
+  public abstract for(): string;
```

Comment:
RC: The name of this method should be more descriptive about what it's abstracting. It looks like this is expected to return the `resourcePath`, or something similar.